src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts

```diff
@@ -522,4 +534,22 @@ export class ProgrammingExerciseConfigureGradingComponent implements OnInit, OnD
             )
             .subscribe();
     }
+
+    public isSumOfWeightsGreaterThanZero(testCaseUpdates: ProgrammingExerciseTestCaseUpdate[]): boolean {
+        // Copy the existing test cases and update the test cases that were changed
+        const copyTestCases = cloneDeep(this.testCases);
```

Comment:
Not sure how much memory/perf this is taking, but I would guess that doing everything in one loop would save us a bit here (?):
```
let weight = 0;
this.testCases.foreach((testCase) => {
    const index = testCaseUpdates.findIndex((update) => testCase.id === update.id);
    if (index !== -1) {
        weight += testCaseUpdates[index].weight ?? 0;
    } else {
        weight += testCase.weight ?? 0;
    }
});
return weight > 0;
```
(untested)

Reply:
Yep, you are right. Your way is cleaner, thanks! :)

Done in `16df0615c`